packages/chaire-lib-backend/src/services/auth/user.ts

```diff
@@ -58,13 +59,16 @@ export default class UserModel extends bookshelf.Model<UserModel> {
         return bcrypt.hashSync(password, bcrypt.genSaltSync(10));
     };
 
-    static confirmAccount = async (token: string): Promise<'Confirmed' | 'NotFound'> => {
+    static confirmAccount = async (token: string, confirmedByAdmin?: boolean): Promise<'Confirmed' | 'NotFound'> => {
```

Comment:
To avoid a circular dependency, here, you could pass an optional confirmation callback that receives a user in parameter and that is called after the save. It also gives more flexibility to what to do after user confirmation.

Reply:
Ideally, we would have a set of pre registered callbacks instead of passing it to the function, so that no matter where it's called from, the callback is active. But I will not do that now.